packages/spatial/src/renderer/components/MeshComponent.ts

```diff
@@ -48,13 +48,23 @@ export const MeshComponent = defineComponent({
 
 export const sceneMeshQuery = defineQuery([MeshComponent, SourceComponent])
 
+/**
+ *
+ * Adds a MeshComponent to an entity with the passed in Geometry and Material
+ * return the geometry and material wrapped in a state, if the state of the geometry or material is updated it will update the geometry and material for the mesh
+ *
+ * @param entity entity to add the mesh component to
+ * @param geometry a Geometry instance to add to the mesh
+ * @param material a Material instance to add to the mesh
+ * @returns [Mesh, State<Geometry>, State<Material>]
+ */
 export function useMeshComponent<
```

Comment:
why a separate useMeshComponent hook instead of just making this the component reactor for the MeshComponent

Reply:
The typing of the mesh's geometry and material is lost when going directly through components, I think you're right about the logic being scoped to the component reactors and I can make that change, but I think the hook is still useful as type inference wrapper.

Reply:
That is fair, but I am uneasy about this hook still. Hooks should not be mutating data, and the strict binding of geometry component and material component to the mesh component is going to need to be changed soon for GLTF loading anyway, this will make that more dificult. These components should be separable, even though in practice they act together.